Create markers.lua

fusion-develops opened the pull request:
For ESX point system simpler version of ox libs

Gellipapa commented:
Hi! @fusion-develops 

I like your solution and thanks for pr, but I don't know how happy Linden would be to use exactly the same name, so he's really referring to the ox_lib points function. 

If we don't look at this part because we are, say, just drawing inspiration from ox_lib, then we have name convention problems and probably the code doesn't work because it is used somewhere "inside" or "Inside" something in code.

Have you tested this code?

fusion-develops commented:
> Hi! @fusion-develops
> 
> I like your solution and thanks for pr, but I don't know how happy Linden would be to use exactly the same name, so he's really referring to the ox_lib points function.
> 
> If we don't look at this part because we are, say, just drawing inspiration from ox_lib, then we have name convention problems and probably the code doesn't work because it is used somewhere "inside" or "Inside" something in code.
> 
> Have you tested this code?

I've redone it and fixed the bugs and changed the name to Markers since that's what it would mostly be used for once approved I can use this to redo all the add-on resources

Gellipapa commented on [core]/es_extended/client/modules/points.lua:
```diff
@@ -0,0 +1,60 @@
+        local Coords = GetEntityCoords(ESX.PlayerData.Ped)
+        for i = 1, #Markers do
+            local Marker = Markers[i]
+            Marker.distance = #(Coords - Marker.Coords) 
```
Marker.Coords doesn't exist, and you should handle vector3. I mean convert if not vector 3.

fusion-develops replied:
Done

Gellipapa commented on [core]/es_extended/client/modules/points.lua:
```diff
@@ -0,0 +1,60 @@
+            local Marker = Markers[i]
+            Marker.distance = #(Coords - Marker.Coords) 
+            if Marker.distance < Marker.radius then 
+                if Marker.status and Marker.Inside then 
```
Marker.Inside, Marker.inside bad variable name, and don't use. You dont't set Marker.inside.

fusion-develops replied:
Fixed Marker.Inside is now a check if the function exists

Gellipapa commented on [core]/es_extended/client/modules/points.lua:
```diff
@@ -0,0 +1,60 @@
+CreateThread(function()
+    while true do 
+        Wait(0)
+        local InPoint = false
```
Change all variable is correct lowerCamelCase. Thanks

fusion-develops replied:
Removed Variable for a better way to do what it was meant for

Gellipapa reviewed (changes requested):
Please test your code before open PR. Thanks a lot.

Gellipapa commented on [core]/es_extended/client/modules/markers.lua:
```diff
@@ -0,0 +1,101 @@
+            if Marker.distance < Marker.radius then 
+                if Marker.status and Marker.Inside and not Marker.running then 
+                    Marker.running = true
+                    Markers:Run(i)
```
hm, if i have 5 markers i created 5 thread?  I think we should avoid this and run code once thread.

Gellipapa replied:
and please change all variable is correct lowerCamelCase. Thanks

fusion-develops replied:
Fixed both brother

Gellipapa reviewed (commented):
Do you tested this refactored code? Probably the while loop will cause you to stop the rest of the code from running if there is any.

Gellipapa commented:
Closed due to inactivity.
